src/User/Application/CommandHandler/ConfirmPasswordResetCommandHandler.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\User\Application\CommandHandler;
+
+use App\Shared\Domain\Bus\Command\CommandHandlerInterface;
+use App\Shared\Domain\Bus\Event\EventBusInterface;
+use App\User\Application\Command\ConfirmPasswordResetCommand;
+use App\User\Application\Command\ConfirmPasswordResetCommandResponse;
+use App\User\Application\Service\UserTokenMatchValidator;
+use App\User\Domain\Entity\PasswordResetTokenInterface;
+use App\User\Domain\Entity\User;
+use App\User\Domain\Entity\UserInterface;
+use App\User\Domain\Event\PasswordResetConfirmedEvent;
+use App\User\Domain\Repository\PasswordResetTokenRepositoryInterface;
+use App\User\Domain\Repository\UserRepositoryInterface;
+use App\User\Domain\Service\PasswordResetTokenValidatorInterface;
+use Symfony\Component\PasswordHasher\Hasher\PasswordHasherFactoryInterface;
+use Symfony\Component\Uid\Factory\UuidFactory;
+
+final readonly class ConfirmPasswordResetCommandHandler implements
+    CommandHandlerInterface
+{
+    public function __construct(
+        private PasswordResetTokenRepositoryInterface $tokenRepository,
+        private UserRepositoryInterface $userRepository,
+        private PasswordHasherFactoryInterface $hasherFactory,
+        private EventBusInterface $eventBus,
+        private UuidFactory $uuidFactory,
+        private PasswordResetTokenValidatorInterface $tokenValidator,
+        private UserTokenMatchValidator $userTokenMatchValidator,
+    ) {
+    }
+
+    public function __invoke(ConfirmPasswordResetCommand $command): void
+    {
+        $passwordResetToken = $this->getValidatedToken($command->token);
+        $user = $this->userTokenMatchValidator->validateAndGetUser(
+            $passwordResetToken,
+            $command->userId
+        );
+
+        $this->updateUserPassword($user, $command->newPassword);
+        $this->markTokenAsUsed($passwordResetToken);
+        $this->publishEvent($user);
+
+        $command->setResponse(
+            new ConfirmPasswordResetCommandResponse(
+                'Password has been reset successfully.'
```

Comment:
please remove the message, we don't need to return any text to user
it will be enough to send the http code in response for such case